tests/test_edit.rs

```diff
@@ -82,6 +118,50 @@ dc = "eqdc10"
     );
 }
 
+#[test]
+fn test_inserted_leaf_table_goes_after_last_sibling() {
+    given(r#"
+        [package]
+        [dependencies]
+        [[example]]
+        [dependencies.opencl]
+        [dev-dependencies]"#
+    ).running(|root| {
+        root["dependencies"]["newthing"] = table();
+    }).produces_display(r#"
+        [package]
+        [dependencies]
+        [dependencies.opencl]
+
+[dependencies.newthing]
+        [[example]]
+        [dev-dependencies]
+"#).produces_in_original_order(r#"
+        [package]
+        [dependencies]
+        [[example]]
+        [dependencies.opencl]
+
+[dependencies.newthing]
+        [dev-dependencies]
+"#);
+}
+
+#[test]
+fn test_inserting_tables_from_different_parsed_docs() {
+    given(
+        "[a]"
+    ).running(|root| {
+        let other = "[b]".parse::<Document>().unwrap();
+        root["b"] = other["b"].clone();
+    }).produces_display(
+        "[a]\n[b]\n"
+    ).produces_in_original_order(
+        // Known limitation of the current approach: b goes missing when
+        // printing.
+        "[a]\n"
```

Comment:
In this test, `[a]` and `[b]` both have `.position = 1` so only `[a]` gets printed.

Is this enough of a bug that we should switch to the slower re-ordering scheme that I described in the PR description, or is it considered an unsupported edge case?